pkg/planner/core/operator/logicalop/base_logical_plan.go

```diff
@@ -50,6 +55,21 @@ type BaseLogicalPlan struct {
 	fdSet *fd.FDSet
 }
 
+// *************************** implementation of HashEquals interface ***************************
+
+// Hash64 implements HashEquals.<0th> interface.
+func (p *BaseLogicalPlan) Hash64(h mutil.Hasher) {
+	h.HashInt(p.ID())
+}
+
+// Equals implements HashEquals.<1st> interface.
+func (p *BaseLogicalPlan) Equals(other *BaseLogicalPlan) bool {
+	if other != nil {
+		return p.ID() == other.ID()
+	}
+	return false
```

Comment:
I think we can add `intest.Assert(false, "should not be called")`

Reply:
yes, great